eth_hash/backends/pycryptodome.py

```diff
@@ -1,9 +1,32 @@
 from Crypto.Hash import (
     keccak,
 )
+from eth_hash.preimage import (
+    BasePreImage,
+)
 
 
 def keccak256(prehash: bytes) -> bytes:
-    hasher = keccak.new(digest_bits=256)
-    hasher.update(prehash)
+    hasher = keccak.new(data=prehash, digest_bits=256)
     return hasher.digest()
+
+
+class preimage(BasePreImage):
+    _hash = None
+
+    def __init__(self, prehash) -> None:
+        self._hash = keccak.new(data=prehash, digest_bits=256)
```

Comment:
I thought you'd need to pass `update_after_digest=True` here, no? Maybe it is actually needed and we're missing a test that calls `update()` after `digest()`?

Reply:
I think you're right, digging in now, will open a new PR.

Reply:
Looking at this further I'm not finding a reasonable way to make pycryptodome based hashes copyable without keeping a copy of all of the parts around which is not idea.  Given that the `copy()` functionality is the only reason this exists and that the need for the preimage functionality is only present to facilitate the `p2p` package, I'm even more inclined to:

- have the `p2p` package use pysha3
- drop the `preimage` functionality from `eth-hash`.

Reply:
I added this issue to track alternative solutions: https://github.com/ethereum/eth-hash/issues/12

(Mostly because I want to see pypy3 support in py-evm)